lib/vdb/vdb.cpp

```diff
@@ -0,0 +1,275 @@
+                        .transform_error([&input](const auto err) {
+                            return Error{ParseError{std::string(input.cbegin(), err)}};
+                        });
+                }),
+                       "failed to load KEYWORDS")
+                    .value_or(pms_utils::ebuild::keywords{});
+        }
+    }
+
+    {
+        auto iuse_path = path_ / "IUSE";
+
+        if (std::filesystem::exists(iuse_path)) {
+            iuse_ = unwrap(try_readfile(iuse_path).ok_or(Error{IoError{}}).and_then([](const auto &input) {
+                        return try_parse(input, pms_utils::parsers::vdb::_internal::VDB_IUSE, true)
+                            .transform_error([&input](const auto err) {
+                                return Error{ParseError{std::string(input.cbegin(), err)}};
+                            });
+                    }),
+                           "failed to load IUSE")
+                        .value_or(pms_utils::ebuild::iuse{});
+        }
+    }
+
+    {
+        auto use_path = path_ / "USE";
+
+        if (std::filesystem::exists(use_path)) {
+            use_ = unwrap(try_readfile(use_path).ok_or(Error{IoError{}}).and_then([](const auto &input) {
+                       return try_parse(input, pms_utils::parsers::vdb::_internal::use_flags, true)
+                           .transform_error([&input](const auto err) {
+                               return Error{ParseError{std::string(input.cbegin(), err)}};
+                           });
+                   }),
+                          "failed to load USE")
+                       .value_or(std::vector<pms_utils::atom::Useflag>{});
+        }
+    }
+
+    {
+        auto contents_path = path_ / "CONTENTS";
+
+        contents_ =
+            unwrap(try_readfile(contents_path).ok_or(Error{IoError{}}).and_then([](const auto &input) {
+                return try_parse(input, pms_utils::parsers::vdb::contents, true)
+                    .transform_error([&input](const auto err) {
+                        return Error{ParseError{std::string(input.cbegin(), err)}};
+                    });
+            }),
+                   "failed to load CONTENTS")
+                .value_or(std::vector<Content>{});
+    }
+
+    repository_ =
+        unwrap(try_readfile(path_ / "repository").ok_or(Error{IoError{}}), "failed to load repository");
+
+    size_ = unwrap(try_readfile(path_ / "SIZE").ok_or(Error{IoError{}}).and_then([](const auto &input) {
+        return try_parse(input, pms_utils::parsers::vdb::_internal::size, true)
+            .transform_error(
+                [&input](const auto err) { return Error{ParseError{std::string(input.cbegin(), err)}}; });
+    }),
+                   "failed to load SIZE");
+}
+
+Vdb::Vdb(std::filesystem::path path) : path_(std::move(path)) {};
+
+Vdb::Iterator Vdb::begin() const {
+    Iterator iter{std::filesystem::directory_iterator(path_)};
+    iter.category = Category(*iter.it);
+    return iter;
+}
```

Comment:
why does this have a public path ctor if the only valid way to construct it is with a `Category` ?